cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/utils/AESUtils.kt

```diff
@@ -106,12 +107,15 @@ internal class AESDecrypt(private val secretKey: AES256Key) {
 
             // Decrypt and write the data to given outputPath
             encryptedDataSource.cipherSource(cipher).buffer().use { bufferedSource ->
-                val data = bufferedSource.readByteArray()
-                size = data.size.toLong()
-                outputSink.buffer().use {
-                    it.write(data)
-                    it.flush()
+                val source = bufferedSource.inputStream().source()
+                val contentBuffer = Buffer()
+                var byteCount: Long
+                while (source.read(contentBuffer, BUFFER_SIZE).also { byteCount = it } != -1L) {
+                    size += byteCount
+                    outputSink.write(contentBuffer, byteCount)
+                    outputSink.flush()
```

Comment:
@MohamadJaara this did the trick on the decrypt part =) now we are able to download without issues, tested until 97 MB. We can meet again, so we discuss the writing of temp files and avoid possible collisions